### PR TITLE
fix(dependabot): fix tough-cookie security vulnerability

### PR DESCRIPTION
**Pull request by thaddmt**
#### Description of changes
- Fixes dependabot alerts for `tough-cookie`
  - https://github.com/aws-amplify/amplify-ui/security/dependabot/97
  - https://github.com/aws-amplify/amplify-ui/security/dependabot/96
 - package `@cypress/request` has a `~` dependency on `tough-cookie@2.5.0` which will spit out a warning, they have an issue to fix this - https://github.com/cypress-io/request/issues/31

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -28735,24 +28735,16 @@ toposort@^2.0.2:
   resolved "https://registry.yarnpkg.com/toposort/-/toposort-2.0.2.tgz#ae21768175d1559d48bef35420b2f4962f09c330"
   integrity sha512-0a5EOkAUp8D4moMi2W8ZF8jcga7BgZd91O/yabJCFY8az+XSzeGyTKs0Aoo897iV1Nj6guFq8orWDS96z91oGg==
 
-tough-cookie@^4.0.0, tough-cookie@^4.1.2:
-  version "4.1.2"
-  resolved "https://registry.npmjs.org/tough-cookie/-/tough-cookie-4.1.2.tgz#e53e84b85f24e0b65dd526f46628db6c85f6b874"
-  integrity sha512-G9fqXWoYFZgTc2z8Q5zaHy/vJMjm+WV0AkAeHxVCQiEB1b+dGvWzFW6QV07cY5jQ5gRkeid2qIkzkxUnmoQZUQ==
+tough-cookie@4.1.3, tough-cookie@^4.0.0, tough-cookie@^4.1.2, tough-cookie@~2.5.0:
```
Comment (zchenwei) [inline yarn.lock:28738]: Looks like one of the dependencies uses a v2. Would it be a problem since the fix needs to bump 2 majors?

Comment (reesscot) [reply yarn.lock:28738]: I guess that could depend on whether we use cookies in Cypress?

**Comment by reesscot** [comment]
Can you include the full `yarn why tough-cookie` here?

**Comment by thaddmt** [comment]
> Can you include the full `yarn why tough-cookie` here?

=> Found "tough-cookie@4.1.3"
info Reasons this module exists
   - "_project_#jsdom" depends on it
   - Hoisted from "_project_#jsdom#tough-cookie"
   - Hoisted from "_project_#@aws-amplify#ui-docs#cypress#@cypress#request#tough-cookie"
   - Hoisted from "_project_#@aws-amplify#ui-angular-mono#jest-preset-angular#jest-environment-jsdom#jsdom#tough-cookie"

**Review by wlee221 — APPROVED** [review]

**Pull request merged**